pkg/lib/packagekit.es6

```diff
@@ -264,3 +264,83 @@ export function watchRedHatSubscription(callback) {
             }
         );
     }
+
+/* Support for installing optional packages.
+ *
+ * First call check_optional_packages to determine whether something
+ * needs to be installed, then call install_optional_packages to
+ * actually install them.
+ *
+ * check_optional_packages resolves to either null when no packages
+ * need to be installed, or an object that can be passed to
+ * install_optional_packages.
+ *
+ * In the future, that object might contain more information about
+ * how many packages need to be installed, and how much space that
+ * will take, but, PackageKit...
+ */
+
+export function check_optional_packages(names, progress_cb) {
+    if (names.length == 0)
+        return cockpit.resolve(null);
+
+    function refresh() {
+        return cancellableTransaction("RefreshCache", [ false ], progress_cb);
+    }
+
+    function resolve() {
+        var ids = [ ];
+        return cancellableTransaction("Resolve",
+                                      [ Enum.FILTER_ARCH |
+                                        Enum.FILTER_NOT_SOURCE |
+                                        Enum.FILTER_NEWEST,
+                                        names
+                                      ],
+                                      progress_cb,
+                                      {
+                                          Package: (info, package_id) => {
+                                              console.log("P", info, package_id);
+                                              var repos = package_id.split(";")[3].split(":");
```

Comment:
Don't use `var` in arrow functions (`const` or `let`), `var` has broken scoping.

Reply:
Fixed.  But is there something special with `var` in arrow functions?  I think it's not anymore broken with arrow functions than with `function` functions.